crates/paimon/src/error.rs

```diff
@@ -52,6 +52,14 @@ pub enum Error {
         display("Paimon hitting invalid config: {}", message)
     )]
     ConfigInvalid { message: String },
+    #[snafu(
+        visibility(pub(crate)),
+        display("Paimon hitting unexpected avro error {}: {:?}", message, source)
+    )]
+    AvroFailed {
```

Comment:
I believe users don't need to be aware of `AvroFailed`. How about using `DataInvalid` or `DataUnexpected` until we determine which is more appropriate?

Reply:
It's better, updated.